components/board/pi/impl/board.go

```diff
@@ -102,6 +103,8 @@ type piPigpio struct {
 	interruptsHW map[uint]ReconfigurableDigitalInterrupt
 	logger       logging.Logger
 	isClosed     bool
+
+	activeBackgroundWorkers sync.WaitGroup
```

Comment:
We add and remove values from this WaitGroup, but we never wait for them all to finish. When closing the board, I suspect we should wait for everything to shut down.